### PR TITLE
[BREAKING] Next Major Release / Port to BabylonJS

### PR DESCRIPTION
**Pull request by me4502**
Tracks progress of the port to BabylonJS - this will be the next major release of SchematicWebViewer.

TODO:
- [x] Re-add arrow and bar rendering
- [x] Improve performance
- [x] Massive cleanups to the model loading system
- [ ] Ensure UV mapping is working
- [x] Investigate material caching (is it needed?)
- [x] Fix issue where only one instance can render per page
- [x] Make a decision around "loading view" - is watching blocks load in acceptable or does it need a proper loading screen?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,15 +33,14 @@ The final argument is an options object that allows configuring various settings
 
 ```typescript
 interface SchematicRenderOptions {
-    size: string; // The size of the canvas viewport
+    size?: number; // Force the size of the canvas viewport, otherwise uses canvas size
```
Comment (octylFractal) [inline README.md]:
```suggestion
    size?: number; // Force the size of the canvas viewport, if not present use canvas size
```

File: tsconfig.json [hunk 2]
```diff
@@ -13,5 +13,8 @@
         "allowJs": true,
         "declaration": true
     },
-    "exclude": ["node_modules", "dist"]
+    "exclude": ["node_modules", "dist"],
+    "types": [
+        "babylonjs"
+    ]
```
Comment (octylFractal) [inline tsconfig.json:19]: Why is this needed?

**Review by octylFractal — CHANGES_REQUESTED** [review]

File: tsconfig.json (hunk 2 above)
Comment (me4502) [reply tsconfig.json:19]: Babylon requires this.

**Review by octylFractal — APPROVED** [review]
